hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java

```diff
@@ -66,24 +66,17 @@
      */
     public static final EvictionPolicy DEFAULT_EVICTION_POLICY = EvictionPolicy.LRU;
 
-    protected int size = DEFAULT_MAX_ENTRY_COUNT;
-    protected MaxSizePolicy maxSizePolicy = DEFAULT_MAX_SIZE_POLICY;
-    protected EvictionPolicy evictionPolicy = DEFAULT_EVICTION_POLICY;
-
-    protected String comparatorClassName;
-    protected EvictionPolicyComparator comparator;
-
-    /**
-     * Used by the {@link NearCacheConfigAccessor} to
-     * initialize the proper default value for on-heap maps.
-     */
-    boolean sizeConfigured;
+    private MaxSizePolicy maxSizePolicy = DEFAULT_MAX_SIZE_POLICY;
+    private EvictionPolicy evictionPolicy = DEFAULT_EVICTION_POLICY;
+    private String comparatorClassName;
+    private EvictionPolicyComparator comparator;
+    private int size = -1;
```

Comment:
minor: Instead of `-1`, can we use a constant like `UNSET` for the sake of readability.